app/src/examples/RuntimeTests/ReanimatedRuntimeTestsRunner/Matchers.ts

```diff
@@ -22,64 +32,121 @@ export class Matchers {
+    const name = color(currentValue.name, 'green');
+    const threadName = color('UI thread', 'cyan');
+    const expected = color(times, 'green');
+    const received = color(callsCount, 'red');
+
+    return {
+      pass: callsCount === times,
+      message: `Expected ${name} to be called ${expected} times on ${threadName}, but was called ${received} times`,
+      messageNegated: `Expected ${name} NOT to be called ${expected} times on ${threadName}`,
+    };
+  };
+
+  private static _toBeCalledJSMatcher: MatcherFunction = (
+    currentValue: TestValue,
+    times = 1
+  ) => {
+    Matchers._assertValueIsCallTracker(currentValue);
+    Matchers._assertValueIsCallTracker(currentValue);
+    const callsCount = currentValue.onJS;
+    const name = color(currentValue.name, 'green');
+    const threadName = color('JS thread', 'cyan');
+    const expected = color(times, 'green');
+    const received = color(callsCount, 'red');
+
+    return {
+      pass: callsCount === times,
+      message: `Expected ${name} to be called ${expected} times on ${threadName}, but was called ${received} times`,
+      messageNegated: `Expected ${name} NOT to be called ${expected} times on ${threadName}`,
+    };
+  };
 
-  public toBeCalledJS(times = 1) {
-    this.assertValueIsCallTracker(this.currentValue);
-    if (this.currentValue.onJS !== times) {
-      const name = color(this.currentValue.name, 'green');
-      const threadName = color('UI thread', 'cyan');
-      const expected = color(times, 'green');
-      const received = color(this.currentValue.onUI, 'red');
-      this.testCase.errors.push(
-        `Expected ${name} to be called ${expected} times on ${threadName}, but was called ${received} times`
+  private makeThrowingMatcher(matcher: MatcherFunction, isNot = false) {
+    return (expectedValue: TestValue, ...args: Array<unknown>) => {
+      const { pass, message, messageNegated } = matcher(
+        this._currentValue,
+        expectedValue,
+        ...args
       );
-    }
+      if ((!pass && !isNot) || (pass && isNot)) {
+        this._testCase.errors.push(
+          isNot && messageNegated ? messageNegated : message
+        );
+      }
+    };
   }
 
-  public toMatchSnapshot(expectedSnapshots: Array<Record<string, unknown>>) {
-    const capturedSnapshots = this.currentValue as Array<
+  public toBe = this.makeThrowingMatcher(Matchers._toBeMatcher);
+  public toBeCalled = this.makeThrowingMatcher(Matchers._toBeCalledMatcher);
+  public toBeCalledUI = this.makeThrowingMatcher(Matchers._toBeCalledUIMatcher);
+  public toBeCalledJS = this.makeThrowingMatcher(Matchers._toBeCalledJSMatcher);
+
+  public not = {
+    toBe: this.makeThrowingMatcher(Matchers._toBeMatcher, true),
+    toBeCalled: this.makeThrowingMatcher(Matchers._toBeCalledMatcher, true),
+    toBeCalledUI: this.makeThrowingMatcher(Matchers._toBeCalledUIMatcher, true),
+    toBeCalledJS: this.makeThrowingMatcher(Matchers._toBeCalledJSMatcher, true),
+  };
```

Comment:
I think we can do it simpler by create getter for property `not`
```js
get not() {
  this._negation = true;
  return this;
}
```
In this case, we don't need to reassign methods to class field.